runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,316 @@
+
+class StatusBarsOverlay extends Overlay
+{
+	private static final Color DEPLETED = new Color(25, 25, 25, 150);
+	private static final Color PRAYER_COLOR = new Color(50, 200, 200, 200);
+	private static final Color QUICK_PRAYER_COLOR = new Color(57, 255, 186, 225);
+	private static final Color BACKGROUND = new Color(0, 0 , 0, 150);
+	private static final Color HEALTH_COLOR = new Color(200, 35, 0, 150);
+	private static final Color POISONED_COLOR = new Color(0, 145, 0, 150);
+	private static final Color VENOMED_COLOR = new Color(0, 65, 0, 150);
+	private static final Color HEAL_COLOR = new Color (255, 112, 6, 150);
+	private static final Color PRAYER_HEAL_COLOR = new Color(57, 255, 186, 75);
+	private static final BufferedImage NO_IMAGE = new BufferedImage(17, 17, BufferedImage.TYPE_INT_ARGB);
+	private static final int HEIGHT = 252;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private static final int IMAGE_SIZE = 16;
+	private static final int HEALTH_LOCATION_X = 2;
+	private static final int PRAYER_LOCATION_X = 3;
+	private final Client client;
+	private final StatusBarsConfig config;
+	private final SkillIconManager skillIconManager;
+	private final QueryRunner queryRunner;
+	private final ItemStatChanges itemStatChanges;
+	private final TextComponent textComponent = new TextComponent();
+	private Viewport curViewport;
+	private Widget curWidget;
+
+	@Inject
+	private StatusBarsOverlay(Client client, StatusBarsConfig config, SkillIconManager skillIconManager, QueryRunner queryRunner, ItemStatChanges itemStatChanges)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.skillIconManager = skillIconManager;
+		this.queryRunner = queryRunner;
+		this.itemStatChanges = itemStatChanges;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		// get current viewport
+		for (Viewport viewport : Viewport.values())
+		{
+			Widget viewportWidget = client.getWidget(viewport.getViewport());
+			if (viewportWidget != null && !viewportWidget.isHidden())
+			{
+				curViewport = viewport;
+				curWidget = viewportWidget;
+			}
+		}
+
+		// if current viewport == null or hidden then don't render the method.
+		if (curViewport == null || curWidget.isHidden())
+		{
+			return null;
+		}
+
+		// set offsets to the respective viewport
+		Point offsetLeft = curViewport.getOffsetLeft();
+		Point offsetRight = curViewport.getOffsetRight();
+		Point location = curWidget.getCanvasLocation();
+		int height = HEIGHT;
+		int offsetPrayerX = (location.getX() - offsetRight.getX());
+		int offsetHealthX = (location.getX() - offsetLeft.getX());
+		int offsetHealthY = (location.getY() - offsetLeft.getY());
+		int offsetPrayerY = (location.getY() - offsetRight.getY());
+
+		if (curViewport == Viewport.RESIZED_BOTTOM)
+		{
+			offsetPrayerY = (location.getY() - 18 - offsetRight.getY());
```

Comment:
```
offsetPrayerY -= 18;
offsetHealthY -= 18;
```
?

Reply:
Also might be worth making that a constant somewhere so it's not a magic number.